### DISCUSSION TITLE
How to add syntax highlighting to the editor

### DISCUSSION THREAD

**Discussion by bgeneto** [open-webui/open-webui#6691] [Q&A]
Hello. I'm missing syntax highlighting for several languages (mainly PHP) in markdown triple ticks code blocks. How do I add them to the editor.?What framework did you use (scintilla compatible)?

**Comment by gkjpettet** [comment]
I don't have an answer but I too would like to know if there is a way to add additional languages to the syntax highlighting engine built into WebUI. I interact with my LLM to help me code in a proprietary language (Xojo) but it doesn't highlight code fences for it. I have written syntax definitions for most of the popular syntax highlighters for this language but I don't see a way to extend the engine.
